processing/src/main/java/org/apache/carbondata/processing/util/CarbonDataProcessorUtil.java

```diff
@@ -622,4 +624,30 @@ public static String trimErrorMessage(String input) {
     }
     return errorMessage;
   }
+  /**
+   * The method returns true is either logger is enabled or action is redirect
+   * @param configuration
+   * @return
+   */
+  public static boolean isRawDataRequired(CarbonDataLoadConfiguration configuration) {
+    boolean isRawDataRequired = Boolean.parseBoolean(
+        configuration.getDataLoadProperty(DataLoadProcessorConstants.BAD_RECORDS_LOGGER_ENABLE)
+            .toString());
+    // if logger is disabled then check if action is redirect then raw data will be required.
+    if (!isRawDataRequired) {
```

Comment:
Is isRawDataRequired is needed only in the case of LoggerAction.REDIRECT.  In that scenario we can turn ON isRawDataRequired whenever LoggerAction.REDIRECT is true. Why to keep a separate configuration?

Reply:
It is needed for the cases bad record logger is enabled or action is redirect.
No new configuration introduce? Just preparing a flag from existing configuration.